### ISSUE TITLE
expose parameters on ApiRequest to set HTTP timeout values

### ISSUE THREAD

**Issue by asgs** [ndustrialio/contxt-sdk-java#1]
Currently, there doesn't seem to be a way to set the HTTP Connect and Read timeouts.

**Comment by asgs** [comment]
@zachary-kuhn yes, we're using `performRequest()`, so we'll need a hook to set the timeouts

**Comment by zachary-kuhn** [comment]
OK, I have a PR open on another repo with a fix for this. Hoping to get a new JAR out soon.

**Comment by asgs** [comment]
Alright, ping here once the jar is released. will the major/minor version will be incremented?

**Comment by asgs** [comment]
@zachary-kuhn is there an update?

**Comment by zachary-kuhn** [comment]
Yes! Hey, the JAR is published to Sonatype at https://oss.sonatype.org/content/repositories/snapshots/io/ndustrial/contxt-sdk/1.0-SNAPSHOT/. We are still publishing snapshots, but need to prioritize in the future to change the build for this project in order to version releases. The new code changes the use of the libraries so you can pass in options like this:

```java
final ServiceOptions options = ServiceOptions.Builder()
    .connectionTimeout(30000)
    .readTimeout(60000)
    .build();
final Contxt contxt = new Contxt(CLIENT_ID, CLIENT_SECRET, options)
```

If you need to set these values per request and not per service, let me know and we can change the implementation.
